streamer/src/nonblocking/recvmmsg.rs

```diff
@@ -36,6 +38,21 @@ pub async fn recv_mmsg(
     Ok(i)
 }
 
+/// Reads the exact number of packets required to fill `packets`
+pub async fn recv_mmsg_exact(
+    socket: &UdpSocket,
+    packets: &mut [Packet],
+) -> io::Result</*num packets:*/ usize> {
+    let total = packets.len();
+    let mut remaining = total;
+    while remaining != 0 {
+        let first = total - remaining;
+        let res = recv_mmsg(socket, &mut packets[first..]).await?;
```

Comment:
This looks wrong if recv_mmsg doesn't receive all the packets on the first call...

Reply:
Whoops, did not have enough caffeine, didn't see line 49.